src/sentry/similarity/features.py

```diff
@@ -124,13 +123,14 @@ def record(self, events):
                     log = (
                         logger.debug
                         if isinstance(error, self.expected_encoding_errors)
-                        else functools.partial(logger.warning, exc_info=True)
+                        else logger.warning
                     )
                     log(
                         "Could not encode features from %r for %r due to error: %r",
                         event,
                         label,
                         error,
+                        exc_info=True,
```

Comment:
I changed these to match the call above -- this ~technically now includes exception info for `log.debug` but otherwise is equivalent.  mypy was unhappy about the union between `logger.debug` and the `partial` because it's not possible to have a consistent call shape between those two

(same below)